hibernate-core/src/main/java/org/hibernate/dialect/temptable/TemporaryTable.java

```diff
@@ -216,8 +216,9 @@ public static TemporaryTable createIdTable(
 										}
 										final ModelPart fkTarget = keyDescriptor.getTargetPart();
 										if ( !fkTarget.isEntityIdentifierMapping() ) {
-											final Value value = entityBinding.getSubclassProperty( pluralAttribute.getAttributeName() )
-													.getValue();
+											final PersistentClass declaringClass = runtimeModelCreationContext.getBootModel()
+													.getEntityBinding( ( (EntityMappingType) pluralAttribute.getDeclaringType() ).getEntityName() );
```

Comment:
`pluralAttribute.getDeclaringType()` might be an embeddable, which brings me to the next possible problem, we are not considering embeddables in this code block. We need to use something like `org.hibernate.query.sqm.mutation.internal.SqmMutationStrategyHelper#visitCollectionTables` here instead of `entityDescriptor.visitSubTypeAttributeMappings`

Reply:
Thanks for the tip, I've updated the solution with similar logic for plural attributes and embeddables - LMK what you think.